### PR TITLE
OSD-2744 Added webhook validation for updating namespaces

### PR DESCRIPTION
**Pull request by csheremeta**
https://issues.redhat.com/browse/OSD-2744

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ IMAGETAG ?= latest
 CABUNDLECONFIGMAP ?= webhook-cert
 VWC_ANNOTATION ?= managed.openshift.io/inject-cabundle-from
 
-IMG ?= quay.io/lseelye/python3-webhookbase
+IMG ?= quay.io/cshereme/python3-webhookbase
```
Comment (csheremeta) [inline Makefile]: @lisa where should we be pushing these images?

File: src/webhook/namespace_validation.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  
+  try:
+    body_dict = request.json['request']
+    requester_group_memberships = body_dict['userInfo']['groups']
+    if "dedicated-admins" in requester_group_memberships:
+      requested_ns = body_dict['namespace']
+      privileged_namespace_re = '(^kube-.*|^openshift.*|^ops-health-monitoring$|^management-infra$|^default$|^logging$|^sre-app-check$|^redhat-.*)'
+      # match will return a match object if the namespace matches the regex, or None if the namespace doesn't match the regex
+      if re.match(privileged_namespace_re, requested_ns) is not None:
+        return responses.response_deny(req=body_dict, msg="You cannot update the privileged namespace {}.".format(requested_ns))
```
Comment (csheremeta) [inline src/webhook/namespace_validation.py]: @rogbas is this verbiage ok?

File: Makefile (hunk 1 above)
Comment (lisa) [reply Makefile]: @csheremeta I believe the changes made by @robotmaxtron to the build scripts (eg in build/*.sh) handle where the images end up. In many ways the Makefile definition is obsoleted now.

File: templates/10-namespace-validation.ValidatingWebhookConfiguration.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    name: namespace-validation.managed.openshift.io
+    rules: 
+      - operations: 
+          - "UPDATE"
+        apiGroups: 
+          - ""
+        apiVersions: 
+          - "*"
+        resources: 
+          - "namespaces"
```
Comment (rogbas) [inline templates/10-namespace-validation.ValidatingWebhookConfiguration.yaml.tmpl]: Could we please have a Cluster `scope` specified here, just like we have for the group-validation webhook?

File: src/webhook/namespace_validation.py (hunk 2 above)
Comment (rogbas) [reply src/webhook/namespace_validation.py]: Verbiage is good @csheremeta thanks!

File: templates/10-namespace-validation.ValidatingWebhookConfiguration.yaml.tmpl (hunk 3 above)
Comment (csheremeta) [reply templates/10-namespace-validation.ValidatingWebhookConfiguration.yaml.tmpl]: done!

Comment (rogbas) [reply templates/10-namespace-validation.ValidatingWebhookConfiguration.yaml.tmpl]: Probably we could get away without it, but standards-wise it's better to keep the same approach.

I checked and the default scope is value "*", which could potentially cause some issues.

**Pull request merged**